src/main/java/com/example/petbuddybackend/service/chat/ChatService.java

```diff
@@ -79,15 +78,17 @@ public ChatMessageDTO createMessage(Long chatId, String principalEmail, ChatMess
                 chatRoom.getClient().getAccountData() :
                 chatRoom.getCaretaker().getAccountData();
 
-        return chatMapper.mapToChatMessageDTO(createMessage(chatRoom, sender, chatMessage.getContent()));
+        return chatMapper.mapToChatMessageDTO(createMessageConvertTimeZone(chatRoom, sender, chatMessage.getContent()));
     }
-
     @Transactional
-    public ChatMessageDTO createMessage(Long chatId, String principalEmail, ChatMessageSent chatMessage, Role role, ZoneId timeZone) {
-        return convertTimeZone(
-                createMessage(chatId, principalEmail, chatMessage, role),
-                timeZone
-        );
+    public ChatMessageDTO createMessageConvertTimeZone(
```

Comment:
```suggestion
    public ChatMessageDTO createMessag(e
```

Jezeli teraz nie rozrozniamy 2 metod z Zoned i bez to mozna tak zosstawic. Reszta review pozniej ;)